notary-commons/src/main/kotlin/sidechain/iroha/ReliableIrohaChainListener.kt

```diff
@@ -58,23 +65,31 @@ class ReliableIrohaChainListener(
      * Returns an observable that emits a new block every time it gets it from Iroha
      */
     override fun getBlockObservable(
-    ): Result<Observable<Pair<iroha.protocol.BlockOuterClass.Block, () -> Unit>>, Exception> {
+    ): Result<Observable<Pair<BlockOuterClass.Block, () -> Unit>>, Exception> {
         return Result.of {
-            val source = PublishSubject.create<Delivery>()
+            val source = PublishSubject.create<Pair<BlockOuterClass.Block, Long>>()
             val deliverCallback = { consumerTag: String, delivery: Delivery ->
                 // This code is executed inside consumerExecutorService
-                source.onNext(delivery)
-            }
-            val obs: Observable<Pair<iroha.protocol.BlockOuterClass.Block, () -> Unit>> = source.map { delivery ->
                 val block = iroha.protocol.BlockOuterClass.Block.parseFrom(delivery.body)
-                logger.info { "New Iroha block from RMQ arrived. Height ${block.blockV1.payload.height}" }
-                Pair(block, {
+                //TODO shall we ignore too old blocks?
+                if (ableToHandleBlock(block)) {
+                    source.onNext(Pair(block, delivery.envelope.deliveryTag))
+                } else {
+                    logger.warn { "Not able to handle Iroha block ${block.blockV1.payload.height}" }
                     if (!autoAck) {
-                        channel.basicAck(delivery.envelope.deliveryTag, false)
-                        logger.info { "Iroha block delivery confirmed" }
+                        confirmDelivery(delivery.envelope.deliveryTag)
```

Comment:
we can't handle a block but we confirm it's delivered? I don't see any sense, help me

Reply:
We have to acknowledge block delivery anyway. Otherwise, mistakenly sent block will freeze `ReliablieIrohaChainListener`.